src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,170 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.exception.IndexMismatchException;
+import duke.exception.IndexOutOfBoundException;
+import duke.task.Task;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+/**
+ * A class that represents a list of tasks.
+ */
+public class TaskList {
+    /**
+     * The list of tasks.
+     */
+    private final ArrayList<Task> taskList;
+
+    /**
+     * Constructs a {@code TaskList}.
+     *
+     * @param taskList The list of tasks.
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Returns the number of tasks.
+     *
+     * @return The number of tasks.
+     */
+    public int size() {
+        return this.taskList.size();
```

Comment:
I like that you wrote a header comment, but I believe it is not necessary for simple getter methods